app/src/contexts/surveyContext.tsx

```diff
@@ -0,0 +1,84 @@
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import useDataLoader, { DataLoader } from 'hooks/useDataLoader';
+import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
+import React, { createContext, PropsWithChildren, useEffect, useMemo } from 'react';
+import { useParams } from 'react-router';
+
+/**
+ * Context object that stores information about a survey
+ *
+ * @export
+ * @interface ISurveyContext
+ */
+export interface ISurveyContext {
+  /**
+   * The Data Loader used to load survey data
+   *
+   * @type {DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>}
+   * @memberof ISurveyContext
+   */
+  surveyDataLoader: DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>;
+
+  /**
+   * The project ID belonging to the current survey
+   *
+   * @type {number}
+   * @memberof ISurveyContext
+   */
+  projectId: number;
+
+  /**
+   * The ID belonging to the current survey
+   *
+   * @type {number}
+   * @memberof ISurveyContext
+   */
+  surveyId: number;
+}
+
+export const SurveyContext = createContext<ISurveyContext>({
+  surveyDataLoader: {} as DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>,
+  projectId: -1,
+  surveyId: -1
+});
+
+export const SurveyContextProvider = (props: PropsWithChildren<Record<never, any>>) => {
+  const biohubApi = useBiohubApi();
+  const surveyDataLoader = useDataLoader(biohubApi.survey.getSurveyForView);
+  const urlParams = useParams();
+
+  if (!urlParams['id']) {
+    throw new Error(
+      "The project ID found in SurveyContextProvider was invalid. Does your current React route provide an 'id' parameter?"
+    );
+  } else if (!urlParams['survey_id']) {
```

Comment:
(not really an issue) You can just make this a regular if, rather than an else if, since only one of these ifs can throw at a time.